pkg/kclient/secrets_test.go

```diff
@@ -0,0 +1,76 @@
+package kclient
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	ktesting "k8s.io/client-go/testing"
+)
+
+func TestCreateTLSSecret(t *testing.T) {
+
+	tests := []struct {
+		name          string
+		secretName    string
+		componentName string
+		host          string
+		wantErr       bool
+	}{
+		{
+			name:          "Case: Valid secret name",
+			componentName: "testComponent",
+			secretName:    "testComponent-tlssecret",
+			host:          "1.2.3.4.nip.io",
+			wantErr:       false,
+		},
+		{
+			name:          "Case: Invalid secret name",
+			secretName:    "",
+			componentName: "testComponent",
+			host:          "1.2.3.4.nip.io",
+			wantErr:       true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// initialising the fakeclient
+			fkclient, fkclientset := FakeNew()
+			fkclient.Namespace = "default"
+
+			fkclientset.Kubernetes.PrependReactor("create", "secrets", func(action ktesting.Action) (bool, runtime.Object, error) {
+				if tt.secretName == "" {
```

Comment:
we can implement this empty check in the function itself. That way we save a API call and remove the create prependReactor entirely.

Reply:
updated

Reply:
you forgot to update this

Reply:
Missed this one. 
Actually the `secretName` would never be empty, since it's `componentName + "-tlssecret"`. So I checked for invalid componentName instead.